agent-antagonist/src/main.rs

```diff
@@ -0,0 +1,349 @@
+    ///
+    /// - delete the region
+    ///
+    /// Additionally, one task is spawned that will:
+    ///
+    /// - get a list of regions
+    ///
+    /// - for each region, get a list of snapshots for that region
+    Run {
+        /// Address of the crucible agent - leave blank to autodetect if in the
+        /// crucible zone
+        #[clap(short, long)]
+        agent: Option<SocketAddr>,
+
+        /// Dataset for the crucible agent - leave blank to autodetect if in the
+        /// crucible zone
+        #[clap(short, long)]
+        dataset: Option<String>,
+
+        /// How many tasks to spawn performing antagonist loop
+        #[clap(short, long)]
+        tasks: usize,
+    },
+}
+
+fn command(log: &Logger, bin: &'static str, args: &[&str]) -> Result<String> {
+    let start = std::time::Instant::now();
+    let cmd = Command::new(bin).args(args).output()?;
+    let elapsed = start.elapsed();
+    info!(log, "{} {:?} took {:?}", bin, args, elapsed);
+
+    if !cmd.status.success() {
+        bail!("zfs list failed!");
+    }
+
+    Ok(String::from_utf8(cmd.stdout)?.trim_end().to_string())
+}
+
+fn get_client(agent: &SocketAddr) -> CrucibleAgentClient {
+    let dur = std::time::Duration::from_secs(60);
+
+    let client = reqwest::ClientBuilder::new()
+        .connect_timeout(dur)
+        .build()
+        .unwrap();
+
+    CrucibleAgentClient::new_with_client(&format!("http://{}", agent), client)
+}
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let args = Args::try_parse()?;
+    match args {
+        Args::Run {
+            agent,
+            dataset,
+            tasks,
+        } => {
+            let log = build_logger();
+
+            let agent = if let Some(agent) = agent {
+                info!(log, "using agent at argument {:?}", agent);
+                agent
+            } else {
+                info!(log, "finding listen_addr from smf");
+                let listen_addr = command(
+                    &log,
+                    "/usr/sbin/svccfg",
+                    &["-s", "agent:default", "listprop", "config/listen_addr"],
+                )?;
+                let listen_addr =
+                    listen_addr.split(' ').last().unwrap().to_string();
+                info!(log, "using listen_addr {:?}", listen_addr);
+
+                info!(log, "finding listen_port from smf");
+                let listen_port = command(
+                    &log,
+                    "/usr/sbin/svccfg",
+                    &["-s", "agent:default", "listprop", "config/listen_port"],
+                )?;
+                let listen_port =
+                    listen_port.split(' ').last().unwrap().to_string();
+                info!(log, "using listen_port {:?}", listen_port);
+
+                format!("[{}]:{}", listen_addr, listen_port).parse()?
+            };
+
+            let dataset = if let Some(dataset) = dataset {
+                info!(log, "using dataset at argument {:?}", dataset);
```

Comment:
Also Nit, using dataset {:?} maybe?  Maybe for the other one too?

Reply:
removed, see 1137af4